packages/mf-utilities/src/preprocessIonizations.ts

```diff
@@ -0,0 +1,35 @@
+import { MF } from 'mf-parser';
+import type { AtomsMap } from 'mf-parser';
+
+export interface Ionization {
+  mf: string;
+  em: number;
+  charge: number;
+  atoms: AtomsMap;
+}
+
+export function preprocessIonizations(
+  ionizationsString: string | Ionization[] = '',
+): Ionization[] {
+  if (Array.isArray(ionizationsString)) return ionizationsString;
+  const ionizations = ionizationsString.split(/ *[\t\n\r,.;]+ */);
```

Comment:
```suggestion
export function preprocessIonizations(
  ionizationsString: string = '',
): Ionization[] {
  const ionizations = ionizationsString.split(/ *[\t\n\r,.;]+ */);
```

@lpatiny I think it's not a great practice to have this kind of pattern (accept input or result) in which case you need this, should we keep it or it's possible to remove it.

Reply:
Usually the rule for @lpatiny is that if tests do not fail after a change, you can make the change.

Reply:
When I use it in the browser I would rather have a string while if I use it as a library I will rather use an array so you would break some code if you change it (and hopefully so tests).

Reply:
This "feature" is used by other packages covered by tests so I will not change it in this PR. But I think it's an anti-pattern.

Reply:
In general it is problematic to make breaking changes in this library because I have over 10 applications that depends on it and there are no testcases.

Reply:
>  if I use it as a library I will rather use an array

So if you use it as a library, you just have to not use this function. because if you give it an array (even if it's not an array of Ionization) it's an identity function, it will return the exact same array.